### PR TITLE
Link targets with platform specific libs

### PR DESCRIPTION
**Pull request by tomlankhorst**
The 'platform' codebase might require linking with some shared libs / frameworks of the OS. I've tried to centralize this in a function such that it's also easy applied to tests and the like. 

This gets test working on MacOS when using CMake.

### CODE REVIEW

File: src/openrct2-cli/CMakeLists.txt [hunk 1]
```diff
@@ -15,21 +16,13 @@ set (PROJECT openrct2-cli)
 project(${PROJECT} CXX)
 add_executable(${PROJECT} ${OPENRCT2_CLI_SOURCES})
 ipo_set_target_properties(${PROJECT})
-
 target_link_libraries(${PROJECT} "libopenrct2")
+target_link_platform_libraries(${PROJECT})
```
Comment (AaronVanGeffen) [inline src/openrct2-cli/CMakeLists.txt]: I think this gets invoked twice in this file now.

**Comment by tomlankhorst** [comment]
Yes ! That was a fix in the closed PR actually . I'll change this .

**Comment by AaronVanGeffen** [comment]
Appreciate the latest clean-up commit! It's still not linking on macOS for me, though:

```
[231/231] Linking CXX executable openrct2-cli
FAILED: openrct2-cli 
: && /Library/Developer/CommandLineTools/usr/bin/c++  -fstrict-overflow -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Wall -Wextra -Wshadow -Wno-unknown-pragmas -Wno-missing-braces -Wno-comment -Wnonnull -Wno-unused-parameter -DDEBUG=0 -Wno-error=objc-method-access -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/openrct2-cli.dir/src/openrct2-cli/Cli.cpp.o  -o openrct2-cli  libopenrct2.a -framework Cocoa -ljansson -lpng16 -lz -lzip -ldl -lcurl /usr/local/Cellar/openssl/1.0.2q/lib/libssl.dylib /usr/local/Cellar/openssl/1.0.2q/lib/libcrypto.dylib /usr/local/lib/libicuuc.dylib -lfreetype && :
ld: library not found for -ljansson
```

If I manually an `-L /usr/local/lib` to the command, `jansson` is found and linked.

Again, this is not a regression of this PR, but if you have any ideas as to how to fix this...

**Comment by tomlankhorst** [comment]
@AaronVanGeffen CMake did find `jansson `, right?

**Comment by AaronVanGeffen** [comment]
@tomlankhorst Yes, and it's able to link it just fine for `openrct`, just not `openrct2-cli`.

**Comment by fafrd** [comment]
I encountered this same issue today building openrct2 (branch develop, commit e9bcad19) on Mac OS 10.15 Catalina:

```
[ 73%] Linking CXX executable openrct2-cli
ld: library not found for -ljansson
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [openrct2-cli] Error 1
make[1]: *** [CMakeFiles/openrct2-cli.dir/all] Error 2
make: *** [all] Error 2
```

I worked around this by appending `-L /usr/local/lib` to `./CMakeFiles/openrct2-cli.dir/link.txt`

**Pull request merged**